### ISSUE TITLE
Python gevent install fail

### ISSUE THREAD

**Issue by magician333** [termux/termux-packages#283]
I use pip install gevent,but it fail.I have installed libev,libev-dev,libevent,libevent-dev.But it still failed

**Comment by magician333** [comment]
# This is wrong message




Collecting gevent
  Using cached gevent-1.1.1.tar.gz
Requirement already satisfied (use --upgrade to upgrade): greenlet>=0.4.9 in /data/data/com.termux/files/usr/lib/python3.5/site-packages (from gevent)
Installing collected packages: gevent
  Running setup.py install for gevent ... error
    Complete output from command /data/data/com.termux/files/usr/bin/python -u -c "import setuptools, tokenize;__file__='/data/data/com.termux/files/usr/tmp/pip-build-7pfd2l59/gevent/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /data/data/com.termux/files/usr/tmp/pip-37u6dhym-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-aarch64-3.5
    creating build/lib.linux-aarch64-3.5/gevent
    copying gevent/__init__.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/_corecffi_build.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/_fileobjectcommon.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/_fileobjectposix.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/_semaphore.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/_socket2.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/_socket3.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/_socketcommon.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/_ssl2.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/_ssl3.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/_sslgte279.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/_tblib.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/_threading.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/_util_py2.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/backdoor.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/baseserver.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/builtins.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/core.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/corecffi.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/coros.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/event.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/fileobject.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/greenlet.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/hub.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/local.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/lock.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/monkey.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/os.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/pool.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/pywsgi.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/queue.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/resolver_ares.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/resolver_thread.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/select.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/server.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/signal.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/socket.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/ssl.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/subprocess.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/thread.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/threading.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/threadpool.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/timeout.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/util.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/win32util.py -> build/lib.linux-aarch64-3.5/gevent
    copying gevent/wsgi.py -> build/lib.linux-aarch64-3.5/gevent
    running build_ext
    generating cffi module 'build/temp.linux-aarch64-3.5/gevent._corecffi.c'
    creating build/temp.linux-aarch64-3.5
    Running '(cd  "/data/data/com.termux/files/usr/tmp/pip-build-7pfd2l59/gevent/libev"  && /bin/sh ./configure   && cp config.h "$OLDPWD" ) > configure-output.txt' in /data/data/com.termux/files/usr/tmp/pip-build-7pfd2l59/gevent/build/temp.linux-aarch64-3.5/libev
    /data/data/com.termux/files/usr/bin/sh: 1: /bin/sh: not found
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/data/data/com.termux/files/usr/tmp/pip-build-7pfd2l59/gevent/setup.py", line 516, in <module>
        run_setup(ext_modules, run_make=run_make)
      File "/data/data/com.termux/files/usr/tmp/pip-build-7pfd2l59/gevent/setup.py", line 502, in run_setup
        **setup_kwds
      File "/data/data/com.termux/files/usr/lib/python3.5/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/data/data/com.termux/files/usr/lib/python3.5/distutils/dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "/data/data/com.termux/files/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/data/data/com.termux/files/usr/lib/python3.5/site-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/data/data/com.termux/files/usr/lib/python3.5/distutils/command/install.py", line 539, in run
        self.run_command('build')
      File "/data/data/com.termux/files/usr/lib/python3.5/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/data/data/com.termux/files/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/data/data/com.termux/files/usr/lib/python3.5/distutils/command/build.py", line 135, in run
        self.run_command(cmd_name)
      File "/data/data/com.termux/files/usr/lib/python3.5/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/data/data/com.termux/files/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/data/data/com.termux/files/usr/lib/python3.5/site-packages/cffi/setuptools_ext.py", line 109, in run
        base_class.run(self)
      File "/data/data/com.termux/files/usr/lib/python3.5/distutils/command/build_ext.py", line 338, in run
        self.build_extensions()
      File "/data/data/com.termux/files/usr/lib/python3.5/distutils/command/build_ext.py", line 447, in build_extensions
        self._build_extensions_serial()
      File "/data/data/com.termux/files/usr/lib/python3.5/distutils/command/build_ext.py", line 472, in _build_extensions_serial
        self.build_extension(ext)
      File "/data/data/com.termux/files/usr/tmp/pip-build-7pfd2l59/gevent/setup.py", line 305, in build_extension
        self.gevent_prepare(ext)
      File "/data/data/com.termux/files/usr/tmp/pip-build-7pfd2l59/gevent/setup.py", line 302, in gevent_prepare
        configure(self, ext)
      File "/data/data/com.termux/files/usr/tmp/pip-build-7pfd2l59/gevent/setup.py", line 196, in configure_libev
        rc = _system(libev_configure_command)
      File "/data/data/com.termux/files/usr/tmp/pip-build-7pfd2l59/gevent/setup.py", line 172, in _system
        return check_call(cmd, shell=True)
      File "/data/data/com.termux/files/usr/lib/python3.5/subprocess.py", line 584, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '(cd  "/data/data/com.termux/files/usr/tmp/pip-build-7pfd2l59/gevent/libev"  && /bin/sh ./configure   && cp config.h "$OLDPWD" ) > configure-output.txt' returned non-zero exit status 127

    ----------------------------------------
Command "/data/data/com.termux/files/usr/bin/python -u -c "import setuptools, tokenize;__file__='/data/data/com.termux/files/usr/tmp/pip-build-7pfd2l59/gevent/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /data/data/com.termux/files/usr/tmp/pip-37u6dhym-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /data/data/com.termux/files/usr/tmp/pip-build-7pfd2l59/gevent/

**Comment by magician333** [comment]
@fornwall

**Comment by fornwall** [comment]
The problem is that setup scripts for gevent assumes that `/bin/sh` does not exist, which it doesn't on Android. I've created an issue for gevent at https://github.com/gevent/gevent/pull/809.

However, I've managed to install gevent by making gevent avoid building embedded libraries, and instead link to the system ones. Install the deps with:

    apt install make libev-dev c-ares-dev gcc python-dev

then install gevent with:

    EMBED=0 pip install gevent

Does this work for you?
